src/test/unit/controllers/OtpLoginController.test.ts

```diff
@@ -53,12 +53,27 @@ describe('Otp Login Controller', () => {
   it('should render same page if no otp code is entered', () => {
     const otpLoginController = new OtpLoginController();
 
-    const response = { redirect: function() {return '';}} as unknown as Response;
+    const response = { render: function() {return '';}} as unknown as Response;
     const request = { body: {}} as unknown as Request;
 
     const responseMock = sinon.mock(response);
 
-    responseMock.expects('redirect').once().withArgs('otp-login');
+    responseMock.expects('render').once().withArgs('otp-login');
+
+    otpLoginController.post(request, response);
+
+    responseMock.verify();
+  });
+
+  it('should render same page if otp code is not digits', () => {
+    const otpLoginController = new OtpLoginController();
+
+    const response = { render: function() {return '';}} as unknown as Response;
+    const request = { body: { 'otp-login': 'abcdef'}} as unknown as Request;
+
+    const responseMock = sinon.mock(response);
+
+    responseMock.expects('render').once().withArgs('otp-login');
```

Comment:
youll need to add more tests in here to cover the multiple scenarios youve introduced with various ifs and else ifs